src/handlers/notebook.ts

```diff
@@ -46,28 +48,32 @@ export class DebuggerNotebookHandler implements IDisposable {
     }
     this.isDisposed = true;
     this.cellManager.dispose();
-    this.notebookTracker.activeCellChanged.disconnect(this.onNewCell);
     Signal.clearData(this);
   }
 
-  protected onNewCell(notebookTracker: NotebookTracker, codeCell: CodeCell) {
-    if (this.cellManager) {
-      this.cellManager.activeCell = codeCell;
-    } else {
-      this.cellManager = new CellManager({
-        breakpointsModel: this.breakpoints,
-        activeCell: codeCell,
-        debuggerModel: this.debuggerModel,
-        debuggerService: this.debuggerService,
-        type: 'notebook'
-      });
-    }
+  protected onNewCell(noteTracker: NotebookTracker, codeCell: CodeCell) {
+    requestAnimationFrame(() => {
+      if (noteTracker.currentWidget.id === this.id) {
```

Comment:
What this means is that we would get the `activeCellChanged` signal here, sent from the notebook tracker, but the cell is from another notebook widget.

And the `cellManager` is created for one notebook widget.